source/common/http/http1/codec_impl.cc

```diff
@@ -346,6 +346,8 @@ void ResponseEncoderImpl::encodeHeaders(const ResponseHeaderMap& headers, bool e
 
   // The contract is that client codecs must ensure that :status is present.
   ASSERT(headers.Status() != nullptr);
+  // encodeHeaders() is not supposed to throw any exceptions. This will crash on a release assert if
+  // response status is invalid.
```

Comment:
may also say "The contract is that client codecs will ensure that the :status value is valid. This will crash otherwise."
(and @yanavlasov should this be ASSERT or crash with RELEASE_ASSERT like now?)

Reply:
Changed string. From below, seems like he's okay with RELEASE_ASSERT :)!